### PR TITLE
Fix EndpointMetadataCollection filename

### PR DESCRIPTION
**Pull request by JamesNK**
https://github.com/aspnet/Routing/issues/527

### CODE REVIEW

File: src/Microsoft.AspNetCore.Routing.Abstractions/DispatcherDataSource.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public abstract class DispatcherDataSource : IEndpointCollectionProvider
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Routing.Abstractions/DispatcherDataSource.cs]: I left out `Address` for now. I need to learn more about the concept of addresses.

**Comment by JamesNK** [comment]
@rynowak Are there abstractions required that I have missed?

**Comment by rynowak** [comment]
what do you think we gain by moving this type to abstractions?

**Comment by JamesNK** [comment]
Ah, I mistakenly thought that MVC referenced only the abstractions so it had to be here.

Hmmm, right now I don't know enough about the use and purpose of the abstractions package to say where it should go.

**Comment by rynowak** [comment]
Yeah I'm honestly not sure either.

**Comment by rynowak** [comment]
> what do you think we gain by moving this type to abstractions?

To help explore this a little more - **if** the data source is part of abstractions - then we're effectively saying "you can provide routing info by just referencing routing.abstractions". I'm not sure if that's going to shake out to be true, or untrue.

----

My perspective (I also want to hear yours): I not sure if it's a priority to enable `you can provide routing info by just referencing routing.abstractions`. Should we let other things happen and see if it comes up? I don't want to make a bunch of 2nd order and 3rd order decisions based on something unimportant.

It's also non-breaking to move a type from routing -> routing.abstractions.

**Pull request merged**